fedimint-cli/src/lib.rs

```diff
@@ -279,9 +282,22 @@ impl Opts {
         Ok(WsAdminClient::new(url, *our_id, auth))
     }
 
-    fn load_config(&self) -> CliResult<ClientConfig> {
-        let cfg_path = self.workdir()?.join("client.json");
-        load_from_file(&cfg_path).map_err_cli_msg(CliErrorKind::IOError, "could not load config")
+    async fn load_config(&self) -> CliResult<ClientConfig> {
+        let db = Database::new(self.load_rocks_db()?, Default::default());
+        let mut dbtx = db.begin_transaction().await;
+
+        let cfg = dbtx
+            .find_by_prefix(&ClientConfigKeyPrefix)
+            .await
+            .next()
+            .await
+            .ok_or_cli_msg(CliErrorKind::GeneralFailure, "no saved config found in db")?
+            .1;
+        dbtx.commit_tx_result()
```

Comment:
You don't need to commit if the transaction is only doing a read

Reply:
@m1sterc001guy, how do I ensure the DB is freed? I have a similar scenario where I don't know how to free and move the DB resource after initializing a read-only dbtx
```rs
let (config, decoders, db) = {
  let db = client.inner.db.clone();
  let decoders = client.inner.decoders.clone();

  // Read config from DB
  let mut dbtx = db.begin_transaction().await;
  let config = dbtx
    .find_by_prefix(&ClientConfigKeyPrefix)
    .await
    .next()
    .await
    .ok_or(anyhow!("No client config found in database"))?
    .1;
  dbtx.commit_tx().await;

  (config, decoders, db)
}
 ```

Reply:
Do you want to free the db or the database transaction? If you just want to free the transaction, you should just be able to let it go out of scope.

Reply:
Nice